### ISSUE TITLE
Protect accessing Compass using a Gate

### ISSUE THREAD

**Issue by Propaganistas** [davidhsianturi/laravel-compass#28]
It would be great if Compass could support checking a Gate permission before allowing access, similar to how Horizon and Telescope define it:

```php
Gate::define('viewCompass', function ($user) {
    return true;
});
```

https://laravel.com/docs/6.x/horizon#dashboard-authorization
https://laravel.com/docs/6.x/telescope#dashboard-authorization

**Comment by Propaganistas** [comment]
A homegrown solution can of course be implemented when Compass supports middleware (see #24 )

**Comment by davidhsianturi** [comment]
No plans to support this at the moment. #24
